### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by dieum512**
In this milestone I:

- [x] Set up associations between models.
- [x] Wrote SQL queries with Active Record.
- [x] Created model classes for all entities as shown in the ERD diagram
- [x] Set up associations between models.
- [x] Used models to insert data
- [x] Createed custom methods

Thanks for the time to correct this work

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
```
Comment (TadesseDev) [inline app/models/user.rb:4]: - [x] Dear @dieum512, for consistency and to stay aligned with the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), Please apply the `foreign_key: 'author_id'` for the `Comment` and `Like` associations too.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (TadesseDev) [inline app/models/comment.rb]: - [x] Here also, the association for the comment should be using a `author_id` column. Hence, Please apply `belongs_to :author, class_name: 'User'`

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (TadesseDev) [inline app/models/like.rb]: - [x] Here also, the association for the Like should be using a `author_id` column. Hence, Please apply `belongs_to :author, class_name: 'User'`

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @dieum512 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @dieum512, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Gitflow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**